CIPs/CIP-71/CIP-71.md

```diff
@@ -0,0 +1,184 @@
+---
+cip: 71
+title: MultiQueries
+author: Joel Thorstensson (@oed)
+discussions-to: https://github.com/ceramicnetwork/CIP/issues/72
+status: Draft
+category: Standards
+type: Core
+created: 2020-11-06
+---
+
+## Simple Summary
+
+A specification for how to query multiple documents at once.
+
+
+## Abstract
+
+This CIP introduces a new way of querying documents in Ceramic. The *CeramicApi* gets an update to support this. The http daemon get's a new endpoint which supports the new query format, and the core gossip protocol get's a revamp with new features. The main feature that these changes allow for is to query multiple documents by specifying a a number of paths. If a path includes a Ceramic URL both the main document as well as the linked document will be synced.
+
+
+## Motivation
+
+Ceramic is a network of linked documents and currently you can only sync one document at a time. This means that if you have a data structure where many documents are linked and you want to sync all of them you would need to make multiple requests to the network. By introducing an api and a gossip specification that allows queries for multiple documents at once this problem is mitigated.
+
+
+## Specification
+
+There are three main changes outlined below. The most important one is **Pubsub queries** which describes the protocol change required for this entire CIP. 
+
+### CeramicApi
+
+The *CeramicApi* is updated to be able to sync mutliple documents (preform a Multiquery) that may be linked from the initial document thoughout the `paths`.
+
+```typescript
+function loadDocuments(id: DocID, paths: string[]): Promise<Record<DocID, Doctype>>
```

Comment:
should we rename a method as discussed earlier? the name might be misleading - we can use something like `loadDocumentPaths` or similar

Reply:
Was it `loadLinkedDocuments` we wanted to use?

Reply:
yeah, I think it was 👍